client/mysqldump.c

```diff
@@ -7069,20 +7150,37 @@ int main(int argc, char **argv)
           "name for any table or database.\n", argv[argument]);
       }
     }
-
-    if (argc > 1 && !opt_databases)
-    {
-      /* Only one database and selected table(s) */
-      if (!opt_alltspcs && !opt_notspcs)
-        dump_tablespaces_for_tables(*argv, (argv + 1), (argc - 1));
-      dump_selected_tables(*argv, (argv + 1), (argc - 1));
+    if (opt_wildcards)
+    {
+      int i= 0;
+      if (argc > 1 && !opt_databases)
+        for (i= 1; i < argc; i++)
+          dump_tablespaces_for_database_wild(argv[0], argv[i]);
+      /*one database, tables matching the wildcard*/
```

Comment:
this is rather unconventional. Normally the comment is **before** the code. please, move